onnxruntime/core/providers/cpu/text/string_normalizer.cc

```diff
@@ -319,95 +486,154 @@ Status StringNormalizer::Compute(OpKernelContext* ctx) const {
   using namespace string_normalizer;
 
   auto X = ctx->Input<Tensor>(0);
-  if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
   auto input_dims = X->Shape().GetDims();
 
-  size_t N = 0;
-  size_t C = 0;
+  TensorShapeVector output_shape;
+  int64_t C = 0;
   if (input_dims.size() == 1) {
     if (input_dims[0] < 1) {
       return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
                     "Single dimension value must be greater than 0");
     }
-    C = onnxruntime::narrow<size_t>(input_dims[0]);
+    C = input_dims[0];
   } else if (input_dims.size() == 2) {
     if (input_dims[0] != 1 || input_dims[1] < 1) {
       return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
                     "Input dimensions are either[C > 0] or [1][C > 0] allowed");
     }
-    N = 1;
-    C = onnxruntime::narrow<size_t>(input_dims[1]);
+    output_shape.push_back(1);
+    C = input_dims[1];
   } else {
     return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT,
                   "Input dimensions are either[C > 0] or [1][C > 0] allowed");
   }
 
-  Status status;
+  // Special case, no filtering and no case change
+  if (case_change_action_ == NONE &&
+      ((is_case_sensitive_ && stopwords_.empty()) ||
+       (!is_case_sensitive_ && wstopwords_.empty()))) {
+    output_shape.push_back(C);
+    auto output_tensor = ctx->Output(0, output_shape);
+    auto const output_data = output_tensor->MutableData<std::string>();
+    std::copy(X->Data<std::string>(), X->Data<std::string>() + C, output_data);
+    return Status::OK();
+  }
+
+  // We need to know the result dimension, and for that we need to filter
+  // the words first. If we compare with stop words is case insensitive, we
```

Comment:
not clear to me what this means:
"If we compare with stop words is case insensitive"